ambry-frontend/src/main/java/com/github/ambry/frontend/UserQuotaRequestCostPolicy.java

```diff
@@ -112,7 +114,11 @@ private double calculateCapacityUnitCost(RestRequest restRequest, RestResponseCh
         double cost = Math.ceil(contentSize / CU_COST_UNIT);
         return (cost > 0) ? cost : 1;
       case GET:
-        if (restRequest.getUri().contains(Operations.GET_SIGNED_URL)) {
+        String uri = restRequest.getUri();
+        if (uri.contains(Operations.GET_SIGNED_URL)) {
+          return INDEX_ONLY_COST;
+        } else if (uri.contains(SubResource.BlobInfo.name()) || uri.contains(SubResource.UserMetadata.name())) {
```

Comment:
I originally used:
```
RestUtils.SubResource subResource = getRequestPath(restRequest).getSubResource();
if (subResource == SubResource.BlobInfo || subResource == SubResource.UserMetadata) { ... }
```
But this threw NPE in unit test that uses mock request with no request path.  Does it make more sense to change the tests?

Reply:
Yep, I think it makes sense to change the test to set the `RequestPath` in the mock request. Many of the frontend package tests do this (or the argument is set automatically by the production code in FrontendRestRequestService). It may be good to add a test utility to do this.